cdap-api/src/main/java/co/cask/cdap/api/metadata/MetadataEntity.java

```diff
@@ -26,34 +26,37 @@
 import javax.annotation.Nullable;
 
 /**
- * Entity representation for Metadata. This representation support representing any of the existing
- * CDAP entities which support metadata or any custom CDAP resource/entity. MetadataEntity are just ordered
  * <p>
- * key-value pair which represent the resource/entity.
+ * Represents either a CDAP entity or a custom entity in metadata APIs. A {@code MetadataEntity} is an ordered
+ * sequence of key/value pairs.
+ * </p>
  * <p>
  * Example usage:
- * Creating a MetadataEntity for stream
+ * Creating a MetadataEntity for Dataset
+ * </p>
  * <pre>
- *  MetadataEntity metadataEntity = MetadataEntity.ofNamespace("myNamespace").append(MetadataEntity.STREAM, "myStream");
+ *  MetadataEntity metadataEntity =
+ *    MetadataEntity.ofNamespace("myNamespace").appendAsType(MetadataEntity.DATASET, "myDataset");
  * </pre>
  * <p>
- * Creating a Metadata for a field in dataset which is a custom resource
+ * Creating a {@code MetadataEntity} for a custom entity: a field in a dataset:
+ * </p>
  * <pre>
  * MetadataEntity metadataEntity = MetadataEntity.ofNamespace("myNamespace")
- *   .append(MetadataEntity.DATASET, "myDataset").append("field", "myField");
+ *   .append(MetadataEntity.DATASET, "myDataset").appendAsType("field", "myField");
  * </pre>
  * <p>
- * By default the key of the last key-value pair becomes the type of the MetadataEntity but it is also possible to
- * change the type on demand to represent the MetadataEntity to represent entities/resources whose type are not the key
- * of last key value pair. An example of this is ApplicationId which ends with "version" although the type is
- * application.
+ * Every metadata entity has a type. The type can be changed during append by using
+ * {@link #appendAsType(String, String)} which does not necessarily have to be the last append.
+ * In some cases, the type is a key in middle. An example of this is ApplicationId which ends with "version"
+ * although the type is "application".
+ * </p>
```

Comment:
Is it still true that by default, the last key is the type? In other words, do I always have to use appendAsType(), or is it only cases when the type is not the last key?

Reply:
Discussed this in person.
Summary of discussion: 
append() never changes type. 
appendAsType() changes type for clear understanding of behavior of the APIs.